pkg/virt-controller/watch/migration.go

```diff
@@ -140,15 +140,15 @@ func StartMigrationTargetPod(v services.VMService, migration *v1.Migration) erro
 
 	//TODO:  detect collisions
 	conflicts := []string{}
-	for k, v := range migration.Spec.DestinationNodeSelector {
+	for k, v := range migration.Spec.NodeSelector {
```

Comment:
Node selectors in kuberntes are a hard constraint (affinity is handled somewhere else). The VM itself can have nodeSelectors, they have the highest priority. 

It effectively allows things like host pinning. If we allow overriding them, they are more something like a soft affinity. If you want your VM to be able to run on other hosts, then you shouldn't make it a hard requirement of the VM you try to migrate.

One of the next patches will add event notifications. Then it easy for users to see why it does not migrate. It can then be resolved by removing the conflicting nodeSelectors either from the VM or the migration and try again ...